File: frontend/src/dtos/models_dtos/ProfileDTO.ts
import type { PositionDTO } from "./PositionDTO"
import type { SocialLinkDTO } from "./social_link"
import type { UserDTO } from "./UserDTO"


export class ProfileDTO {

    id: number
    about: string
    bannerUrl: string
    positions: PositionDTO[]
    socialLinks: SocialLinkDTO[]
    user: UserDTO

    constructor(id: number,about: string,bannerUrl: string,positions: PositionDTO[], socialLinks: SocialLinkDTO[]  ,user: UserDTO) {
        this.id = id 
        this.about = about
        this.bannerUrl = bannerUrl
        this.positions = positions
        this.socialLinks = socialLinks
        this.user = user
    }

}

File: backend/src/main/java/com/grouply/backend/profile/dto/ProfileDTO.java
package com.grouply.backend.profile.dto;

import com.grouply.backend.position.dto.PositionDTO;
import com.grouply.backend.profile.social_link.dto.SocialLinkDTO;
import com.grouply.backend.user.Dtos.UserDTO;
import lombok.Builder;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

@Data
@Builder
public class ProfileDTO {

    private Long id;

    private String about;
    private String bannerUrl;

    private Set<SocialLinkDTO> socialLinks;

    private List<PositionDTO> positions = new ArrayList<>();

    private UserDTO user;

}
